src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/IdentityProviderEntityBuilder.java

```diff
@@ -0,0 +1,119 @@
+import com.ca.apim.gateway.cagatewayconfig.tasks.zip.beans.identityprovider.BindOnlyLdapIdentityProviderDetail;
+import com.ca.apim.gateway.cagatewayconfig.tasks.zip.beans.identityprovider.IdentityProvider;
+import com.ca.apim.gateway.cagatewayconfig.util.IdGenerator;
+import com.ca.apim.gateway.cagatewayconfig.util.gateway.BuilderUtils;
+import com.ca.apim.gateway.cagatewayconfig.util.xml.DocumentTools;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class IdentityProviderEntityBuilder implements EntityBuilder {
+    private final Document document;
+    private final IdGenerator idGenerator;
+
+    IdentityProviderEntityBuilder(Document document, IdGenerator idGenerator) {
+        this.document = document;
+        this.idGenerator = idGenerator;
+    }
+
+    public List<Entity> build(Bundle bundle) {
+        return bundle.getIdentityProviders().entrySet().stream().map(identityProviderEntry ->
+                buildIdentityProviderEntity(identityProviderEntry.getKey(), identityProviderEntry.getValue())
+        ).collect(Collectors.toList());
+    }
+
+    private Entity buildIdentityProviderEntity(String name, IdentityProvider identityProvider) {
+        final Element identityProviderElement = document.createElement("l7:IdentityProvider");
+        final String id = idGenerator.generate();
+        identityProviderElement.setAttribute("id", id);
+        identityProviderElement.appendChild(DocumentTools.createElement(document, "l7:Name", name));
+        identityProviderElement.appendChild(DocumentTools.createElement(document, "l7:IdentityProviderType", identityProvider.getType().getValue()));
+
+        switch(identityProvider.getType()) {
+            case BIND_ONLY_LDAP:
+                if (identityProvider.getProperties() != null) {
+                    identityProviderElement.appendChild(BuilderUtils.buildPropertiesElement(
+                            identityProvider.getProperties().entrySet()
+                                    .stream()
+                                    .collect(Collectors.toMap(stringStringEntry -> "property." + stringStringEntry.getKey(), Map.Entry::getValue)), document));
+                }
+                identityProviderElement.appendChild(buildBindOnlyLdapIPDetails(identityProvider));
+                break;
+            case LDAP:
+                //do something
+                if (identityProvider.getProperties() != null) {
+                    identityProviderElement.appendChild(BuilderUtils.buildPropertiesElement(
+                            identityProvider.getProperties().entrySet()
+                                    .stream()
+                                    .collect(Collectors.toMap(stringStringEntry -> "property." + stringStringEntry.getKey(), Map.Entry::getValue)), document));
+                }
+                //TODO: fill out other ldap details
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-developer-plugin&pullRequest=58&issues=AWVKO96liZQY_aXE1Ls1&open=AWVKO96liZQY_aXE1Ls1)

[ISSUE_KEY]: AWVKO96liZQY_aXE1Ls1